pkg/asset/installconfig/azure/azure.go

```diff
@@ -0,0 +1,93 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/Azure/go-autorest/autorest/to"
+
+	"github.com/openshift/installer/pkg/types/azure"
+
+	"github.com/pkg/errors"
+	survey "gopkg.in/AlecAivazis/survey.v1"
+
+	azsub "github.com/Azure/azure-sdk-for-go/profiles/latest/resources/mgmt/subscriptions"
+)
+
+const (
+	defaultRegion string = "eastus"
+)
+
+// Platform collects azure-specific configuration.
+func Platform() (*azure.Platform, error) {
+	regions, err := getRegions()
+	longRegions := make([]string, 0, len(regions))
+	shortRegions := make([]string, 0, len(regions))
+	for id, location := range regions {
+		longRegions = append(longRegions, fmt.Sprintf("%s (%s)", id, location))
+		shortRegions = append(shortRegions, id)
+	}
+	regionTransform := survey.TransformString(func(s string) string {
+		return strings.SplitN(s, " ", 2)[0]
+	})
+
+	_, ok := regions[defaultRegion]
+	if !ok {
+		return nil, errors.Errorf("installer bug: invalid default azure region %q", defaultRegion)
+	}
+
+	sort.Strings(longRegions)
+	sort.Strings(shortRegions)
+
+	var region string
+	err = survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Select{
+				Message: "Region",
+				Help:    "The azure region to be used for installation.",
+				Default: fmt.Sprintf("%s (%s)", defaultRegion, regions[defaultRegion]),
+				Options: longRegions,
+			},
+			Validate: survey.ComposeValidators(survey.Required, func(ans interface{}) error {
+				choice := regionTransform(ans).(string)
+				i := sort.SearchStrings(shortRegions, choice)
+				if i == len(shortRegions) || shortRegions[i] != choice {
+					return errors.Errorf("invalid region %q", choice)
+				}
+				return nil
+			}),
+			Transform: regionTransform,
+		},
+	}, &region)
+	if err != nil {
+		return nil, err
+	}
+
+	return &azure.Platform{
+		Region: region,
+	}, nil
+}
+
+func getRegions() (map[string]string, error) {
+	ctx := context.TODO()
+	session, err := GetSession()
+	if err != nil {
+		return nil, err
+	}
+	client := azsub.NewClient()
+	client.Authorizer = session.Authorizer
+	locations, err := client.ListLocations(ctx, session.SubscriptionID)
```

Comment:
nit: why not `context.TODO()` here itself.

Reply:
I will update this with a better context, setting an appropriate timeout.